File: backend/src/main/java/com/electoral/entities/Partido.java
package com.electoral.entities;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "partidos")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Partido {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String nombre;

    @Column(length = 20)
    private String sigla;

    @Column(name = "logo_url", length = 500)
    private String logoUrl;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "elecciones_id", nullable = false)
    private Eleccion elecciones;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: conteo_electoral_app/lib/models/partido.dart
class Partido {
  final int id;
  final String nombre;
  final String sigla;
  final String? logoUrl;
  final int eleccionesId;

  Partido({
    required this.id,
    required this.nombre,
    required this.sigla,
    this.logoUrl,
    required this.eleccionesId,
  });

  factory Partido.fromJson(Map<String, dynamic> json) {
    return Partido(
      id: json['id'],
      nombre: json['nombre'],
      sigla: json['sigla'] ?? '',
      logoUrl: json['logoUrl'],
      eleccionesId: json['eleccionesId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombre': nombre,
      'sigla': sigla,
      'logoUrl': logoUrl,
      'eleccionesId': eleccionesId,
    };
  }
}
